virtual_ecosystem/models/abiotic_simple/microclimate.py

```diff
@@ -301,34 +271,28 @@ def interpolate_soil_temperature(
         soil temperature profile, [C]
     """
 
-    # select surface layer (atmosphere)
-    surface_layer = layer_heights[layer_heights.coords["layer_roles"] == "surface"]
-
-    # create array of interpolation heights including surface layer and soil layers
-    interpolation_heights = xr.concat(
-        [
-            surface_layer,
-            layer_heights[layer_heights.coords["layer_roles"] == "soil"] * -1
-            + surface_layer.values,
-        ],
-        dim="layers",
+    # select surface layer (atmosphere) and generate interpolation heights
+    surface_layer = layer_heights[layer_structure.role_indices["surface"]].to_numpy()
+    soil_depths = layer_heights[layer_structure.role_indices["all_soil"]].to_numpy()
```

Comment:
I think we could use the `layer_structure.soil_layer_thickness` here?

Reply:
I think this is equally direct.